File: programmers/src/stage1/java/이상한문자만들기.java
package stage1;

public class 이상한문자만들기 {
	public String solution(String s) {
		char[] ch = s.toCharArray();
		int idx = 0; //인덱스 판별
		for (int i = 0; i < ch.length; i++) {
			if (ch[i] == ' ') { // 공백일 경우 인덱스 0으로 초기화
				idx = 0;
				continue;
			}
			if (idx % 2 == 0) {
				ch[i] = Character.toUpperCase(ch[i]);
			} else {
				ch[i] = Character.toLowerCase(ch[i]);
			}
			idx++;
		}
		return String.valueOf(ch);
    }
}

File: programmers/src/stage1/python/이상한문자만들기.py
def solution(s):
    answer = ""
    # 단어의 index를 저장
    idx = 0
    for i in s:
        # 공백일 때
        if i == " ":
            answer += i
            idx = 0

        # 공백이 아닐 때
        else:
            # 단어의 index가 짝수
            if idx % 2 == 0:
                answer += i.upper()
            # 단어의 index가 홀수
            else:
                answer += i.lower()
            idx += 1
    return answer
